File: deal-intelligence-platform/frontend/types/deal.ts
export type HealthStatus = "Healthy" | "Needs Review" | "At Risk";
export type Momentum = "increasing" | "stable" | "declining";

export type Deal = {
  id: string;
  name: string;
  account: string;
  owner: string;
  deal_stage: string;
  opportunity_type: string;
  deal_value: number;
  probability: number;
  stage_age_days: number;
  inactivity_days: number;
  engagement_score: number;
  meetings_count: number;
  email_count: number;
  competitor_mentioned: boolean;
  next_step_defined: boolean;
  close_date_pushed: boolean;
  sentiment_score: number;
  forecast_trend: Momentum;
  health_status: HealthStatus;
  ai_confidence: number;
  close_date: string;
  drivers: string[];
  next_best_action: string;
  manual_next_step?: string;
  human_review_required?: boolean;
  raw_details?: Record<string, string | number | boolean | null>;
};

export type Interaction = {
  id: string;
  deal_id: string;
  channel: "email" | "call" | "meeting";
  actor: string;
  summary: string;
  sentiment: "positive" | "neutral" | "negative";
  created_at: string;
};

export type RiskPoint = {
  date: string;
  risk: number;
  confidence: number;
};

File: deal-intelligence-platform/backend/fastapi-ml/app/schemas/deal.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

Momentum = Literal["increasing", "stable", "declining"]
HealthStatus = Literal["Healthy", "Needs Review", "At Risk"]


class DealFeatures(BaseModel):
    deal_stage: str
    opportunity_type: str
    deal_value: float = Field(ge=0)
    probability: float = Field(ge=0, le=100)
    stage_age_days: int = Field(ge=0)
    inactivity_days: int = Field(ge=0)
    engagement_score: float = Field(ge=0, le=100)
    meetings_count: int = Field(ge=0)
    email_count: int = Field(ge=0)
    competitor_mentioned: bool
    next_step_defined: bool
    close_date_pushed: bool
    sentiment_score: float = Field(ge=-1, le=1)
    forecast_trend: Momentum


class PredictionResponse(BaseModel):
    prediction: HealthStatus
    confidence: int
    drivers: list[str]
    shap_factors: list[dict]
    next_best_action: str


class SentimentRequest(BaseModel):
    text: str = Field(min_length=1)


class AgentRunRequest(BaseModel):
    deal_id: str | None = None
    payload: dict = Field(default_factory=dict)
